apps/test/unit/templates/teacherNavigation/lessonMaterials/LessonMaterialsContainerTest.tsx

```diff
@@ -132,4 +134,157 @@ describe('LessonMaterialsContainer', () => {
         .length === 0
     );
   });
+
+  describe('resource links', () => {
+    let windowOpenMock: jest.SpyInstance<
+      Window | null,
+      Parameters<typeof utils.windowOpen>
+    >;
+
+    beforeEach(() => {
+      windowOpenMock = jest
+        .spyOn(utils, 'windowOpen')
+        .mockImplementation(() => null);
+    });
+
+    afterEach(() => {
+      windowOpenMock.mockRestore();
+      jest.resetAllMocks();
+    });
+
+    function viewResource(resourceName: string, resourceAction: string) {
+      const label = screen.getByText(resourceName);
+      const resourceRow = label.closest(
+        '[data-testid="resource-row"]'
+      ) as HTMLElement;
+      expect(resourceRow).toBeDefined();
+      const menuButton = resourceRow.querySelector('button') as HTMLElement;
+      expect(menuButton).toBeDefined();
+      fireEvent.click(menuButton);
+
+      const dropdown = menuButton?.nextElementSibling as HTMLElement;
+      expect(dropdown.classList.contains('dropdownMenuContainer')).toBe(true);
+
+      const viewButton = within(dropdown).getByText(
+        resourceAction
+      ) as HTMLElement;
+      fireEvent.click(viewButton);
+    }
```

Comment:
nit: Instead of trying to go through the parent child relationships here, could we just use `screen.getByText(resourceAction)` or 
```
const dropdown = screen.getByTestId('dropdown-id')
within(dropdown).getByText(resourceAction)
```

Reply:
yes, good point! because there should only be one dropdown visible on the screen.